docs/features/search/declarative-integration.md

```diff
@@ -0,0 +1,230 @@
+
+This is a guide for experimenting with `Search` in a declarative integrated Backstage front-end application.
+
+## Main Concepts
+
+Using declarative integration, you can customize your Backstage instance without writing code, see this [RFC](https://github.com/backstage/backstage/issues/18372) for more information.
+
+In the new frontend system, everything that extends Backstage's core features is called an extension, so an extension can be anything from an API to a page component.
+
+Extensions produces output artifacts and these artifacts are inputs consumed by other extensions:
+
+![search extensions example](../../assets/search/search-extensions-example.svg)
+
+In the image above, a `SearchResultItem` extension outputs a component and this component is injected as input to the `SearchPage` "items" attachment point. The `SearchPage` in turn uses the search result items to compose a search page element and outputs a route path and the page element so they are used as inputs attached to the `CoreRoutes` extension. Finally, the `CoreRoutes` renders the page element when the location matches the search page path.
+
+The basic concepts briefly mentioned are crucial to understanding how the declarative version of the `Search` plugin works.
+
+## Search Plugin
+
+The search plugin is a collection of extensions that implement the search feature in Backstage.
+
+### Installation
+
+Only one step is required to start using the `Search` plugin within declarative integration, so all you have to do is to install the `@backstage/plugin-catalog` and `@backstage/plugin-search` packages, (e.g., [app-next](https://github.com/backstage/backstage/tree/master/packages/app-next)):
+
+```sh
+yarn add @backstage/plugin-catalog @backstage/plugin-search
+```
+
+The `Search` plugin depends on the `Catalog API`, that's the reason we have to install the ` @backstage/plugin-catalog` package too.
+
+### Extensions
+
+The `Search` plugin provides the following [extensions preset](https://github.com/backstage/backstage/blob/3f4a44aef39bd8dbf5098e60b6fdf66fd754c6d9/plugins/search/src/alpha.tsx#L246):
+
+- **SearchApi**: Outputs a concrete implementation for the `Search API` that is attached as an input to the `Core` apis holder;
+- **SearchPage**: Outputs a component that represents the advanced `Search` page interface, this extension expects `Search` result items components as inputs to use them for rendering results in a custom way;
+- **SearchNavItem**: It is an extension that outputs a data that represents a `Search` item in the main application sidebar, in other words, it inputs a sidebar item to the `Core` nav extension.
+
+### Configurations
+
+The `Search` extensions are configurable via `app-config.yaml` file in the `app.extensions` field using the extension id as the configuration key:
+
+_Example disabling the search page extension_
+
+```yaml
+# app-config.yaml
+app:
+  experimental:
+    packages: 'all' # enable packages auto discovery
```

Comment:
Thoughts on skipping this? Thinking to slim it down and avoid needing to change it in the future

Reply:
Makes sense, `experimental` config omitted 🙌🏻 : https://github.com/backstage/backstage/pull/20328/commits/8623fdeca87e79e4b1769db84c4482032dbbda9c